packages/output/src/browser/output-contribution.ts

```diff
@@ -33,4 +43,45 @@ export class OutputContribution extends AbstractViewContribution<OutputWidget> {
         });
     }
 
+    registerCommands(commands: CommandRegistry): void {
+        super.registerCommands(commands);
+        commands.registerCommand(OutputCommands.CLEAR_ALL_TOOLBAR, {
+            isEnabled: widget => this.withWidget(widget, () => true),
+            isVisible: widget => this.withWidget(widget, () => true),
+            execute: widget => this.withWidget(widget, outputWidget => this.clear(outputWidget))
+        });
+    }
+
+    async registerToolbarItems(toolbarRegistry: TabBarToolbarRegistry): Promise<void> {
+        toolbarRegistry.registerItem({
+            id: 'channels',
+            render: () => this.withWidget(undefined, widget => widget.renderChannelSelection()),
+            isVisible: widget => {
```

Comment:
@westbury would it be enough to do the following instead?

```ts
isVisible: widget => this.withWidget(widget, () => true)
```

Reply:
This was moved out of the ViewContribution, so has now become just
`isVisible: widget => (widget instanceof OutputWidget)`